packages/wallets/particle/src/adapter.ts

```diff
@@ -1,175 +1,277 @@
+  readonly supportedTransactionVersions = null;
 
-    get publicKey() {
-        return this._publicKey;
-    }
+  private _connecting: boolean;
+  private _wallet: SolanaWallet | null;
+  private _publicKey: PublicKey | null;
+  private _config: ParticleAdapterConfig;
+  private _readyState: WalletReadyState =
+    typeof window === "undefined"
+      ? WalletReadyState.Unsupported
+      : WalletReadyState.Loadable;
 
-    get connecting() {
-        return this._connecting;
-    }
+  private _particleNetwork: ParticleNetwork | null = null;
 
-    get readyState() {
-        return this._readyState;
-    }
+  constructor(config: ParticleAdapterConfig) {
+    super();
+    this._connecting = false;
+    this._publicKey = null;
+    this._wallet = null;
 
-    async connect(): Promise<void> {
-        try {
-            if (this.connected || this.connecting) return;
-            if (this._readyState !== WalletReadyState.Loadable) throw new WalletNotReadyError();
+    const nestedConfig: NestedConfig = config.config;
 
-            this._connecting = true;
+    const chainId =
+      nestedConfig.chainId !== undefined ? nestedConfig.chainId : 101;
+    const chainName =
+      nestedConfig.chainName !== undefined ? nestedConfig.chainName : "solana";
 
-            let ParticleClass: typeof ParticleNetwork;
-            let WalletClass: typeof SolanaWallet;
-            try {
-                ({ ParticleNetwork: ParticleClass, SolanaWallet: WalletClass } = await import(
-                    '@particle-network/solana-wallet'
-                ));
-            } catch (error: any) {
-                throw new WalletLoadError(error?.message, error);
-            }
+    this._config = {
+      ...config,
+      config: {
+        ...nestedConfig,
+        chainId,
+        chainName,
+        projectId: nestedConfig.projectId,
+        clientKey: nestedConfig.clientKey,
+        appId: nestedConfig.appId,
+      },
+    };
 
-            let wallet: SolanaWallet;
+    if (config.preferredAuthType) {
+      if (typeof config.preferredAuthType === 'object' && 'type' in config.preferredAuthType) {
+        const preferredAuthType = config.preferredAuthType as PreferredAuthTypeObject;
+        if (preferredAuthType.setAsDisplay && preferredAuthType.type) {
+          (async () => {
             try {
-                wallet = new WalletClass(new ParticleClass(this._config?.config).auth);
-            } catch (error: any) {
-                throw new WalletConfigError(error?.message, error);
+              this.icon = await import(
+                `./${preferredAuthType.type}.ts`
+              ).then((module) => module.default);
+              console.log(this.icon);
+            } catch (error) {
+              console.error(`Could not import icon: ${error}`);
             }
+          })();
+          this.name = (preferredAuthType.type.charAt(0).toUpperCase() +
+            preferredAuthType.type.slice(1)) as WalletName<any>;
```

Comment:
We're also concerned about this functionality. We can't really have "Github" for example showing in the wallet modal, we need users to be able to trust what is displayed here at least by the wallets that we publish from this repo

It needs to be clear in all cases that the user is connecting to Particle wallet
We also can't use other people's logos here